android/KMEA/app/src/main/java/com/keyman/engine/KMManager.java

```diff
@@ -2137,23 +2137,18 @@ public static boolean updateText(KeyboardType kbType, String text) {
     return result;
   }
 
-  public static boolean updateSelectionRange(KeyboardType kbType, int selStart, int selEnd) {
+  public static boolean updateSelectionRange(KeyboardType kbType) {
```

Comment:
Noting this would be an [API change](https://help.keyman.com/developer/engine/android/17.0/KMManager/updateSelectionRange).


Do we just have  this?
```java
public static boolean updateSelectionRange(KeyboardType kbType, int selStart, int selEnd) {
  return updateSelectionRange(KeyboardType kbType);
}
```

Reply:
Ah, didn't realize that was public API.

We could just deprecate the parameters, I guess?  The engine is now able to retrieve those values from the source.

Or would we need to find some way to make that work?  It doesn't seem to be meaningfully possible to use `KMKeyboard` without a backing `onCreateInputConnection` call, noting this code section:

https://github.com/keymanapp/keyman/blob/93dc41e2f5e1892f0de3fa821f44da8fbd144355/android/KMEA/app/src/main/java/com/keyman/engine/KMManager.java#L472-L481

Both in-app and system keyboards have a very specific connection to their respective text sources.

As a small side question... I don't see where or when we ever _close_ the connection objects built within that block... isn't that asking for a memory leak over time?  I know Java has garbage collection, but I don't think all languages with GC have associated destructor patterns that can resolve and clean up connection stuff like this.

Reply:
I think we go ahead and deprecate the parameters. (dunno  what to do about the potential memory leak).

At some point, we should also update `updateSelectionRange` used in SystemKeyboard for KMSample2 and the FV app

Reply:
Yes, deprecation of parameters is good -- that's backwardly compatible. No changing API signatures in beta!

As for the connection objects, I see no indication at https://developer.android.com/reference/android/view/inputmethod/InputConnection or https://developer.android.com/reference/android/inputmethodservice/InputMethodService#getCurrentInputConnection() that we need to manage cleanup?

Reply:
Maybe true, but isn't it a given (at least, in some programming languages) that connections need cleanup?